src/codeu/chat/server/Model.java

```diff
@@ -160,7 +169,12 @@ public void add(User user, ConversationHeader conversation) {
     return conversationByText;
   }
 
-  public StoreAccessor<Uuid, ConversationPayload> conversationPayloadById() {
+  public StoreAccessor<Uuid, ConversationPayload> conversationPayloadById(Uuid user, Uuid conversation) throws Exception {
+    ConversationHeader conversationServer = conversationById.first(conversation);
```

Comment:
This is not the right place for this logic.  This is just an accessor for the conversationPayloadById object, any logic permissions should be where this method is called, such as the add() function.